File: 2020/06/advent202006.go
package main

import (
	"fmt"
	"os"
	"strings"
)

func main() {
	data, err := os.ReadFile("input.txt")
	if err != nil {
		panic(err)
	}
	input := strings.Split(string(data), "\n")
	fmt.Println(part1(input))
	fmt.Println(part2(input))
}

func findAnswers(data []string) map[rune]int {
	answers := map[rune]int{}
	for _, line := range data {
		for _, char := range line {
			answers[char]++
		}
	}
	return answers
}

func part1(input []string) int {
	row := 0
	sum := 0
	for row < len(input) {
		data := []string{}
		for row < len(input) && len(input[row]) > 0 {
			data = append(data, input[row])
			row++
		}
		sum += len(findAnswers(data))
		row++
	}
	return sum
}

func part2(input []string) int {
	row := 0
	sum := 0
	for row < len(input) {
		data := []string{}
		for row < len(input) && len(input[row]) > 0 {
			data = append(data, input[row])
			row++
		}
		answers := findAnswers(data)
		for _, n := range answers {
			if n == len(data) {
				sum++
			}
		}
		row++
	}
	return sum
}

File: 2020/06/advent202006.py
def part1(lines: list[str]) -> int:
  tmp = ''
  count = 0
  for line in lines:
    if line == '':
      yeses = set(list(tmp))
      count += len(yeses)
      tmp = ''
    else:
      tmp += line
  yeses = set(list(tmp))
  return count + len(yeses)

def part2(lines: list[str]) -> int:
  count = 0
  yeses: dict[str, int] = {}
  members = 0
  for line in lines:
    if line == '':
      for n in yeses.values():
        if n == members:
          count += 1
      yeses: dict[str, int] = {}
      members = 0
    else:
      members += 1
      for c in line:
        if c in yeses:
          yeses[c] += 1
        else:
          yeses[c] = 1
  for n in yeses.values():
    if n == members:
      count += 1
  return count


if __name__ == "__main__":
  file1 = open('input.txt','r')
  lines = [line.strip() for line in file1.readlines()]
  print(part1(lines))
  print(part2(lines))
